src/Test/Utilities/Portable/TestBase.cs

```diff
@@ -147,7 +148,10 @@ public static MetadataReference SystemCoreRef
             {
                 if (s_systemCoreRef == null)
                 {
-                    s_systemCoreRef = AssemblyMetadata.CreateFromImage(TestResources.NetFX.v4_0_30319.System_Core).GetReference(display: "System.Core.v4_0_30319.dll");
+                    Interlocked.CompareExchange(
+                        ref s_systemCoreRef,
+                        AssemblyMetadata.CreateFromImage(TestResources.NetFX.v4_0_30319.System_Core).GetReference(display: "System.Core.v4_0_30319.dll"),
```

Comment:
I am just trying to understand why this is specific to s_systemCoreRef. 
The fix does not fix any potential race other than adding object identity guarantee. But why do we care, and why only for s_systemCoreRef case?